### DISCUSSION TITLE
Is it possible to create kspace from Image

### DISCUSSION THREAD

**Discussion by osbm** [facebookresearch/fastMRI#332] [Q&A]
Hello,

I am trying to learn image reconstruction. I was able to create reconstructions from kspace like this:

```python
kspace = T.to_tensor(kspace)
kspace = fastmri.ifft2c(kspace)
kspace = fastmri.complex_abs(kspace)
kspace = fastmri.rss(kspace, dim=1)
```

The fastmri package converts a complex tensor into a normal float tensor with  a dimension at the end as complex dimension.

I trained an UNet model that fixes subsampled reconstructions. It looks alright.

I want to visually compare the kspaces of generated, ground truth and masked images.  I tried the `fastmri.fft2c` function but it expects data with complex value. But i saved the images after computing `abs` so there is no complex data at hand. Is it even possible? If so how can i do it?

**Comment by mmuckley** [comment]
Hello @osbm, in general you cannot recover complex data after the rss operation, as it takes the magnitude which removes the complex phase. You can try to copy the magnitude data into the real part of a complex number to take the FFT, but this would constitute an inverse crime and carries some heavy limitations. You may want to read this paper: https://www.pnas.org/doi/abs/10.1073/pnas.2117203119.
